### ISSUE TITLE
Add option to limit max triangle edge length

### ISSUE THREAD

**Issue by zdila** [PDAL/PDAL#3924]
**Describe the bug**

Please add option to limit max triangle edge length. I am not sure if it should belong to `filters.delaunay` or to `filters.faceraster`.

Just for the reference - Whitebox Tools has this option - `whitebox_tools -r=LidarDigitalSurfaceModel --max_triangle_edge_length=5 ...`.

**System/installation information:**
```
Linux bono 6.0.0-6-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.0.12-1 (2022-12-09) x86_64 GNU/Linux
```

```
pdal 2.3.0 (git-version: Release)
```

**Comment by hobu** [comment]
see #3928 3928

**Comment by abellgithub** [comment]
I don't think this is doing what you want. This filter creates a raster from an existing triangulation. The edge length is the edge length of the raster cell.

In general, there's no way to limit the edge length of a triangulation of points because points might simply be too far apart to meet the request. Perhaps the OP can refer to an existing algorithm?

**Comment by hobu** [comment]
ah, I was confused on what [edgeLength](https://github.com/PDAL/PDAL/blob/master/filters/FaceRasterFilter.cpp#L82) mean

**Comment by zdila** [comment]
In the existing algorithm it is here: https://github.com/jblindsay/whitebox-tools/blob/e7fcdf3fc6fe71956833723b53afdf445b0608d1/whitebox-tools-app/src/tools/lidar_analysis/lidar_dsm.rs#L634

**Comment by abellgithub** [comment]
That's not a delaunay triangulation. You can't guarantee edge length with a delaunay triangulation.

**Comment by wonder-sk** [comment]
I think the idea was to make `filters.faceraster` optionally ignore triangles where the edge length > X when rasterizing triangles (rather than constraining triangle size in Delaunay triangulation)... The assumption is that triangles with long edge(s) would go across "holes" in point cloud data, and that would be a simple way of filtering them out.

**Comment by zdila** [comment]
Please reopen.

**Comment by hobu** [comment]
If you have a PR to implement this we will consider merging it. I'm not going to try again 😄

**Comment by abellgithub** [comment]
I don't understand this change. What was the goal here? We are now skipping the setting of raster Z's when those points fall inside a triangle with a long edge (and not even setting nodata AFAICT). What is the benefit of that? It adds computation time and can lead to confusion (the option is also not documented).

If you don't like the mesh, the place to fix that is at mesh creation.

**Comment by nicogodet** [comment]
@zdila Is this issue still valid? Or #4114 should have closed this?
I use this option in PDAL wrench (see https://github.com/PDAL/wrench/pull/65)
